ledger/src/shred.rs

```diff
@@ -733,10 +737,6 @@ pub fn should_discard_shred(
                     return true;
                 }
             };
-            if parent_offset == 0 && slot != 0 {
-                stats.bad_parent_offset += 1;
-                return true;
-            }
```

Comment:
In `verify_shred_slots()`, we only allow `parent_offset == 0` if `slot == 0` so this is redundant.